common4j/src/main/com/microsoft/identity/common/java/telemetry/Telemetry.java

```diff
@@ -280,6 +280,38 @@ public void flush(@NonNull final String correlationId) {
         }
     }
 
+    /**
+     * Get the final Telemetry Map for a correlationId.
+     * @param correlationId The correlation id should either passed by the client app through the API call
+     *                      or generated by the API dispatcher.
+     */
+    public List<Map<String, String>> getFinalMap(@NonNull final String correlationId) {
+        if (!mIsTelemetryEnabled) {
+            return Collections.emptyList();
+        }
+
+        if (StringUtil.isNullOrEmpty(correlationId)) {
+            Logger.warn(TAG, "No correlation id set.");
+            return Collections.emptyList();
+        }
+
+        //check the configuration
+        if (!mDefaultConfiguration.isDebugEnabled() && mIsDebugging) {
+            return Collections.emptyList();
+        }
+
+        List<Map<String, String>> finalRawMap = new CopyOnWriteArrayList<>();
+
+        for (Iterator<Map<String, String>> iterator = mTelemetryRawDataMap.iterator(); iterator.hasNext(); ) {
+            Map<String, String> event = iterator.next();
+            if (correlationId.equalsIgnoreCase(event.get(Key.CORRELATION_ID))) {
+                finalRawMap.add(applyPiiOiiRule(event));
+                iterator.remove();
```

Comment:
I don't think you want `remove` here because this would just remove the entry and then when `flush` is called there would be nothing to flush